### PR TITLE
- Uri encoding method added and spec added too

### PR DESCRIPTION
**Pull request by mur-wtag**
[TAPI-536](https://welltravel.atlassian.net/browse/TAPI-536)

### CODE REVIEW

File: lib/rest_in_peace/template_sanitizer.rb [hunk 1]
```diff
@@ -19,7 +19,7 @@ def url
         param = @params.delete(token.to_sym)
         param ||= @klass.send(token) if @klass.respond_to?(token)
         raise IncompleteParams, "No parameter for token :#{token} found" unless param
-        @url.sub!(%r{:#{token}}, param.to_s)
+        @url.sub!(%r{:#{token}}, encode_uri(param.to_s))
```
Comment (luxflux) [inline lib/rest_in_peace/template_sanitizer.rb]: The `to_s` call isn't needed as the guard clause of `encode_uri` handles the case of a `nil` value.

File: lib/rest_in_peace/template_sanitizer.rb [hunk 2]
```diff
@@ -31,5 +31,10 @@ def tokens
     def leftover_params
       @params.delete_if { |param| tokens.include?(param.to_s) }
     end
+
+    def encode_uri(param)
+      return if param.nil?
+      CGI.escape param
```
Comment (luxflux) [inline lib/rest_in_peace/template_sanitizer.rb]: CGI.escape can only handle strings, so we have to call `to_s` on `param` here.

File: spec/rest_in_peace/template_sanitizer_spec.rb [hunk 3]
```diff
@@ -117,6 +117,14 @@
...
         specify { expect { subject }.to raise_error(RESTinPeace::TemplateSanitizer::IncompleteParams) }
       end
+
+      context 'param which need to be encoded' do
+        let(:params) { { id: 'this param need to be encoded' } }
+        let(:encoded_param) { 'this+param+need+to+be+encoded' }
+        let(:attributes) { { a_id: 2 } }
+        let(:url_template) { '/a/:a_id/b/:id' }
+        specify { expect(subject).to eq("/a/2/b/#{encoded_param}") }
+      end
```
Comment (luxflux) [inline spec/rest_in_peace/template_sanitizer_spec.rb]: Please add a spec where the param to be encoded is `nil`.

**Comment by luxflux** [comment]
@calmyournerves IMHO this PR would be ready to merge, what do you think?

**Comment by calmyournerves** [comment]
Thanks! I just pushed version 5.0.1 to Rubygems.

**Comment by luxflux** [comment]
Thanks a lot!

**Pull request merged**